src/main/java/org/folio/marc4ld/service/marc2ld/bib/MarcBib2LdMapperImpl.java

```diff
@@ -49,14 +54,41 @@ public class MarcBib2LdMapperImpl implements MarcBib2ldMapper {
   private final EmptyEdgesCleaner emptyEdgesCleaner;
 
   @Override
-  public Resource fromMarcJson(String marc) {
+  public Optional<Resource> fromMarcJson(String marc) {
     if (isEmpty(marc)) {
       log.warn("Given marc is empty [{}]", marc);
-      return null;
+      return Optional.empty();
+    }
+    var records = marcReaderProcessor.readMarc(marc).filter(this::isMonograph).toList();
+    if (records.isEmpty()) {
+      log.warn("Given marc is not monograph, skipping: [{}]", marc);
+      return Optional.empty();
+    }
+    return Optional.of(createInstanceAndWorkResource(records));
+  }
+
+  private boolean isMonograph(Record record) {
+    var leader = record.getLeader();
+    if (leader == null) {
+      return false;
     }
+    char typeOfRecord = leader.getTypeOfRecord();
+    char bibliographicLevel = leader.getImplDefined1()[0];
+    return isLanguageMaterial(typeOfRecord) && isMonographicComponentPartOrItem(bibliographicLevel);
+  }
+
+  private boolean isLanguageMaterial(char typeOfRecord) {
+    return typeOfRecord == RecordType.LANGUAGE_MATERIAL.value;
```

Comment:
since we now start to use org.folio.marc4ld.service.ld2marc.leader.enums in marc2ld direction, I suggest to move enums package to make it common for both ld2marc & marc2ld directions. org.folio.marc4ld.enums for example

Reply:
done.